### PR TITLE
WIP: Various optimizations

### PR DESCRIPTION
**Pull request by Oaphi**
This utility PR makes various utility optimizations to a bunch of modules

### CODE REVIEW

**Review by samliew — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -1236,15 +1234,15 @@ const announcement = new Announcement();
...
+
         // POST event from /envvar form
         app.post('/envvar', async ({ body }, res) => {
             const { password } = /** @type {{ password:string }} */(body);
 
             const validPwd = password === process.env.PASSWORD;
 
             // Convert request to JSON object - see https://stackoverflow.com/a/8649003
-            const envVars = JSON.parse('{"' + body.replace(/&/g, '","').replace(/=/g,'":"') + '"}', function(key, value) { return key===""?value:decodeURIComponent(value) });
+            const envVars = JSON.parse('{"' + body.replace(/&/g, '","').replace(/=/g, '":"') + '"}', function (key, value) { return key === "" ? value : decodeURIComponent(value); });
```
Comment (samliew) [inline src/index.js:1245]: This has been removed in my latest changes, probably what's causing the conflict.

Comment (Oaphi) [reply src/index.js:1245]: Don't worry, I'll rebase :) Constantly do, so should not be a problem

**Comment by Oaphi** [comment]
Hmm, something's wrong according to CI in terms of fetching ENV vars from Heroku, both on master and after rebasing this branch on top of master too.

**Comment by samliew** [comment]
> Hmm, something's wrong according to CI in terms of fetching ENV vars from Heroku, both on master

I've already added HEROKU_API_TOKEN to this repo's secrets🤔

**Comment by samliew** [comment]
Merging, then splitting the Heroku API stuff into a separate branch

**Pull request merged**